pilot/pkg/xds/eds.go

```diff
@@ -165,9 +168,37 @@ func (s *DiscoveryServer) edsCacheUpdate(shard model.ShardKey, hostname string,
 	// For existing endpoints, we need to do full push if service accounts change.
 	if saUpdated {
 		log.Infof("Full push, service accounts changed, %v", hostname)
-		fullPush = true
+		return true, true
+	}
+	if created {
+		return true, true
+	}
+	// Check if new Endpoints are ready to be pushed. This check
+	// will ensure that if a new pod comes with a non ready endpoint,
```

Comment:
how do you distinguish if it is from a new pod

Reply:
Check the logic below. If it is not there in shards